### PR TITLE
C# support LRO metadata

### LINKED ISSUE #1219: Specify metadata in LRO generated code
Use the new `Operation<TResponse, TMetadata>`.

### PR DESCRIPTION
**Pull request by chrisdunelm**
Fixes #1219

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/csharp_gapic_client_library.baseline [hunk 1]
```diff
@@ -5559,7 +5559,7 @@ namespace Google.Example.Library.V1
             Modify_DiscussBookRequestCallSettings(ref callSettings);
             BidirectionalStreamingSettings effectiveStreamingSettings =
                 streamingSettings ?? _callDiscussBook.StreamingSettings;
-            AsyncDuplexStreamingCall<DiscussBookRequest,Comment> call =
+            AsyncDuplexStreamingCall<DiscussBookRequest,  Comment> call =
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp_gapic_client_library.baseline]: This seems to have gone from no spaces to two spaces... I suspect that wasn't deliberate...

Comment (chrisdunelm) [inline src/test/java/com/google/api/codegen/testdata/csharp_gapic_client_library.baseline]: Hmm,  odd,  I'll have a look

**Comment by chrisdunelm** [comment]
@jskeet PTAL

File: src/test/java/com/google/api/codegen/testdata/csharp_gapic_client_library.baseline (hunk 1 above)
Comment (chrisdunelm) [inline src/test/java/com/google/api/codegen/testdata/csharp_gapic_client_library.baseline]: Fixed

**Pull request merged**